src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1251,7 +1252,7 @@ internal virtual HttpResponseMessage GetResponse(HttpClient client, HttpRequestM
                         CustomMethod = string.Empty;
                     }
 
-                    currentUri = new Uri(request.RequestUri, response.Headers.Location);
+                    currentUri = new Uri(request.RequestUri!, response.Headers.Location);
```

Comment:
The same.

Reply:
I still haven't found a way to correctly handle this, do you have some guidance?

Reply:
If we don't use BaseAddress it is ok
https://source.dot.net/#System.Net.Http/System/Net/Http/HttpClient.cs,751

Reply:
Implemented a possible solution, maybe i could have used `currentUri = response.Headers.Location;`